src/app/dashboard/whitelist/WhiteListPage.js

```diff
@@ -0,0 +1,392 @@
+
+import './style.css'
+
+// might need TableToolbarAction and batchActionClick here
+const { Table, TableBody, TableCell, TableContainer, TableHead, TableHeader, TableRow,
+  TableSelectAll, TableSelectRow, TableToolbar, TableBatchAction, TableBatchActions,
+  TableToolbarSearch, TableToolbarContent } = DataTable
+
+type StateProps = {|
+  transferManager: TransferManager,
+  addresses: Array<string>,
+  sell: Array<number>,
+  buy: Array<number>,
+  investors: Array<Investor>,
+  csvMessage: string,
+  previewCSVShowing: boolean,
+  listLength: number,
+|}
+
+type DispatchProps = {|
+  initialize: () => any,
+  handleUpload: () => any,
+  multiSubmit: () => any,
+  singleSubmit: () => any,
+  getWhitelist: (?Date, ?Date) => any,
+  updateListLength: (number) => any,
+  removeInvestor: (investors: Array<string>) => any,
+  editInvestors: (investors: Array<string>) => any,
+|}
+
+const mapStateToProps = (state) => ({
+  transferManager: state.whitelist.transferManager,
+  addresses: state.whitelist.addresses,
+  sell: state.whitelist.sell,
+  buy: state.whitelist.buy,
+  investors: state.whitelist.investors,
+  csvMessage: state.whitelist.csvMessage,
+  previewCSVShowing: state.whitelist.previewCSVShowing,
+  listLength: state.whitelist.listLength,
+})
+
+const mapDispatchToProps = (dispatch: Function) => ({
+  initialize: () => dispatch(initialize()),
+  handleUpload: (file) => dispatch(uploadCSV(file)),
+  multiSubmit: () => dispatch(multiUserSubmit()),
+  singleSubmit: () => dispatch(oneUserSubmit()),
+  getWhitelist: (calenderStart: Date, calenderEnd: Date) => dispatch(getWhitelist(calenderStart, calenderEnd)),
+  updateListLength: (pageNumber: number) => dispatch(listLength(pageNumber)),
+  removeInvestor: (investors: Array<string>) => dispatch(removeInvestor(investors)),
+  editInvestors: (investors: Array<string>) => dispatch(editInvestors(investors)),
+})
+
+type Props = StateProps & DispatchProps
+
+type State = {
+  page: number,
+  editInvestorsShowing: boolean,
+  editInvestors: Array<string>,
+}
+
+ type EventData = {
+  id: string,
+  address: string,
+  added: ?string,
+  addedBy: ?string,
+  from: string,
+  to: string,
+}
+
+type PageChanger = {
+  page: number,
+  pageSize: number,
+}
+
+type DatePickerType = [Date, Date]
+
+const dateFormat = (date: Date) => date.toLocaleDateString('en', {
+  year: 'numeric',
+  month: 'short',
+  day: 'numeric',
+})
+
+class WhitelistPage extends Component<Props, State> {
+  state = {
+    page: 0,
+    editInvestorsShowing: false,
+    editInvestors: [],
+  }
+
+  componentWillMount () {
+    this.props.initialize()
+  }
+
+  handleChangePages = (pc: PageChanger) => {
+    this.props.updateListLength(pc.pageSize)
+    this.setState({
+      page: (pc.page - 1),
+    })
+  }
+
+  handleDatePicker = (picker: DatePickerType) => {
+    if (picker.length === 2) {
+      this.setState({
+        page: 0, // TODO @davekaj: make sure that reseting to initial page is truly needed
```

Comment:
"reseting" typo
please do not think this is for perfectionism, I'm just worrying about your IDE settings, which allows you to miss typos. they can cause annoying bugs

Reply:
yeah, I am using Atom editor, which I am not used to, because usually I use VScode, but vscode has very bad support for flow. so I don't know many of the atom commands, and i tried to get spell check working and I think it only works on some file types

this typo is fixed now though